src/app/services/alert.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { NavigationStart, Router } from '@angular/router';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class AlertService {
+    private subject = new Subject<any>();
```

Comment:
as suggested before, would be good to create an interface for message and ditch using any